src/components/countries.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import { CountriesContext } from '../contexts/countries';
+import { useSorting, useThemeSwitcher } from '../hooks';
+import { CountriesContextType, Language, TypeCountry } from '../@types/country';
+import { ThemeContext } from '../contexts/theme';
+import { ThemeContextType } from '../@types/theme';
+
+type CountriesProps = {
+    selectedContinent: string | null;
+}
+
+const Countries:React.FC<CountriesProps> = ({ selectedContinent }: CountriesProps) => {
+  const { countries } = React.useContext(CountriesContext) as CountriesContextType;
+  const { theme } = React.useContext(ThemeContext) as ThemeContextType;
+  const [ selectedCountry, setSelectedCountry ] = React.useState<string | null>(null);
+  const wrapperRef = React.useRef<HTMLDivElement | null>(null);
+
+  const continentCountries: TypeCountry[] = countries.filter((country: TypeCountry) => country.continent === selectedContinent)
```

Comment:
```suggestion
  const continentCountries = countries.filter((country: TypeCountry) => country.continent === selectedContinent)
```

The type assertion is probably not necessary, if the data is correctly typed.